supporting_scripts/code-coverage/per_module_cov_report/parse_module_coverage.py

```
@@ -0,0 +1,75 @@
+            total_instructions = instruction_covered + instruction_missed
+            instruction_coverage = (instruction_covered / total_instructions * 100) if total_instructions > 0 else 0.0
+
+            missed_classes = int(class_counter.get('missed', 0))
+
+            results.append({
+                "module": report['module'],
+                "instruction_coverage": instruction_coverage,
+                "missed_classes": missed_classes
+            })
+        except Exception as e:
+            print(f"Error processing {module_folder}: {e}")
```

Ruff (0.8.2)

Undefined name `module_folder`

(F821)